In [1]:
import sys
from pathlib import Path

import pandas as pd

# Add the project root to Python's import path
PROJECT_ROOT = Path.cwd().parents[1]

if str(PROJECT_ROOT) not in sys.path:
    sys.path.append(str(PROJECT_ROOT))

In [2]:
from pathlib import Path

print(Path.cwd())

C:\Users\Manojit Khatua\AI ENGINEERING\trace-ai\src\analysis


In [11]:
from analysis_src.data_inspection import (DataInspector,ViewOfDataStrategy,DataTypesInspectionStrategy,SummaryStatisticalInspectionStrategy,)

In [3]:
import pandas as pd

train_transaction = pd.read_csv("../data/processed/extracted_data/train_transaction.csv")
train_identity = pd.read_csv("../data/processed/extracted_data/train_identity.csv")
print("Train Transaction Shape:", train_transaction.shape)
print("Train Identity Shape:", train_identity.shape)

Train Transaction Shape: (590540, 394)
Train Identity Shape: (144233, 41)


In [8]:
inspector = DataInspector(DataTypesInspectionStrategy())
inspector.execute_inspection(train_transaction)


Data types and non-null information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 590540 entries, 0 to 590539
Columns: 394 entries, TransactionID to V339
dtypes: float64(376), int64(4), object(14)
memory usage: 1.7+ GB


In [19]:
# Summary Statistic 
inspector.execute_inspection(train_identity)
inspector.set_strategy(SummaryStatisticalInspectionStrategy())
print("/n Train_transaction staistic")
inspector.execute_inspection(train_transaction)
print("/n Test_transaction statistic")
inspector.execute_inspection(train_identity)


Numerical summary statistics:
       TransactionID          id_01          id_02         id_03  \
count   1.442330e+05  144233.000000  140872.000000  66324.000000   
mean    3.236329e+06     -10.170502  174716.584708      0.060189   
std     1.788496e+05      14.347949  159651.816856      0.598231   
min     2.987004e+06    -100.000000       1.000000    -13.000000   
25%     3.077142e+06     -10.000000   67992.000000      0.000000   
50%     3.198818e+06      -5.000000  125800.500000      0.000000   
75%     3.392923e+06      -5.000000  228749.000000      0.000000   
max     3.577534e+06       0.000000  999595.000000     10.000000   

              id_04          id_05          id_06        id_07        id_08  \
count  66324.000000  136865.000000  136865.000000  5155.000000  5155.000000   
mean      -0.058938       1.615585      -6.698710    13.285354   -38.600388   
std        0.701015       5.249856      16.491104    11.384207    26.084899   
min      -28.000000     -72.000000    -1

# SUMMARY STATISTIC

# TRAIN TRANSACTION
1. Fraud is strongly imbalanced: Total transactions = 590,540
Fraud ≈ 3.5%
This immediately tells us:Accuracy will be a bad primary metric.
2. TransactionAmt needs investigation ,We see:
median ≈ ₹68.77
75% ≈ ₹125
maximum ≈ ₹31,937
That's a very large spread.
3. TransactionDT is extremely important
min  = 86,400
max  ≈ 15.81 million
4. The V-features are heavily sparse:
while the dataset has 590,540 transactions.
That's only about:82,351 / 590,540 ≈ 13.9%
So roughly 86% are missing for those fields.This is important.We should not automatically delete them.Missingness itself may carry information in fraud datasets.
